src/main/java/com/stuypulse/robot/subsystems/Edwin.java

```diff
@@ -0,0 +1,216 @@
+    }
+
+    // Velocity
+    private double getLeftVelocity() {
+        return leftEncoder.getVelocity();
+    }
+
+    private double getRightVelocity() {
+        return rightEncoder.getVelocity();
+    }
+
+    private double getVelocity() {
+        return (getLeftVelocity() + getRightVelocity()) / 2.0;
+    }
+
+    /***************
+     * ROBOT ANGLE *
+     ***************/
+
+    // Gets current Angle of the Robot as a Rotation2d (contiuous / not +-180)
+    public Rotation2d getRotation2d() {
+        return navx.getRotation2d();
+    }
+
+    /**********************
+     * ODOMETRY FUNCTIONS *
+     **********************/
+
+    private void updateOdometry() {
+        odometry.update(getRotation2d(), getLeftDistance(), getRightDistance());
+    }
+
+    public Pose2d getPose() {
+        return odometry.getPoseMeters();
+    }
+
+    @Override
+    public void setPose(Pose2d pose) {
+        odometry.resetPosition(pose, new Rotation2d());
+    }
+
+    @Override
+    public DifferentialDriveKinematics getKinematics() {
+        return kinematics;
+    }
+
+    @Override
+    public TrajectoryConfig getTrajectoryConfig() {
+        return new TrajectoryConfig(
+            Motion.MAX_VELOCITY, 
+            Motion.MAX_ACCELERATION
+        ).setKinematics(kinematics);
+    }
+
+    /********************
+     * DRIVING COMMANDS *
+     ********************/
+
+    public void stop() {
+        drivetrain.stopMotor();
+    }
+
+    public void drive(double leftMetersPerSecond, double rightMetersPerSecond) {
+        leftTargetSpeed.set(leftMetersPerSecond);
+        rightTargetSpeed.set(rightMetersPerSecond);
+    }
+
+    public void driveVolts(double leftVolts, double rightVolts) {
+        for (CANSparkMax motor : left) {
+            motor.setVoltage(leftVolts);
+        }
+
+        for (CANSparkMax motor : right) {
+            motor.setVoltage(rightVolts);
+        }
+    }
+
+    @Override
+    public void periodic() {
+        updateOdometry();
+        field.setRobotPose(getPose());
+        
+        driveVolts(
+            leftController.update(leftTargetSpeed.get(), getLeftVelocity()), 
+            rightController.update(rightTargetSpeed.get(), getRightVelocity())
+        );
+
+        // LOGGING
+
+        SmartDashboard.putNumber("Edwin/Left Distance", getLeftDistance());
+        SmartDashboard.putNumber("Edwin/Right Distance", getRightDistance());
+        SmartDashboard.putNumber("Edwin/Distance", getDistance());
+
+        SmartDashboard.putNumber("Edwin/Left Vel", getLeftVelocity());
+        SmartDashboard.putNumber("Edwin/Right Vel", getRightVelocity());
+        SmartDashboard.putNumber("Edwin/Velocity", getVelocity());
+
+        SmartDashboard.putNumber("Edwin/Pose X", getPose().getX());
+        SmartDashboard.putNumber("Edwin/Pose Y", getPose().getY());
+        SmartDashboard.putNumber("Edwin/Gyro Angle", getRotation2d().getDegrees());
+    
+        SmartDashboard.putData("Edwin/Field", field);
```

Comment:
This only needs to be called once because putData takes in a Sendable which handles updating itself with lambdas and such.. so move to constructor